nilearn/_utils/niimg_conversions.py

```diff
@@ -33,7 +32,77 @@ def _check_same_fov(img1, img2):
             and np.allclose(img1.get_affine(), img2.get_affine()))
 
 
-def check_niimg(niimg, atleast_4d=False):
+def _index_img(img, index):
+    """Helper function for check_niimg_4d."""
+    return new_img_like(
+        img, img.get_data()[:, :, :, index], img.get_affine(),
+        copy_header=True)
+
+
+def _iter_check_niimg(niimgs, ensure_ndim=None, atleast_4d=False,
+                      target_fov=None,
+                      memory=Memory(cachedir=None),
+                      memory_level=0, verbose=0):
+    """Iterate over a list of niimgs and do sanity checks and resampling
+
+    Parameters
+    ----------
+
+    niimgs: list of niimg
+        Image to iterate over
+
+    ensure_ndim: integer, optional
+        If specified, an error is raised if the data does not have the
+        required dimension.
+
+    atleast_4d: boolean, optional
+        If True, any 3D image is converted to a 4D single scan.
+
+    target_fov: tuple of affine and shape
+       If specified, images are resampled to this field of view
+    """
+    ref_fov = None
+    ndim_minus_one = ensure_ndim - 1 if ensure_ndim is not None else None
```

Comment:
Feel free to propose something else! When I put a 2-line condition, people tell that it's a one-liner ;)